### PR TITLE
Reorganize CLI

### PR DESCRIPTION
**Pull request by AleCandido**
The EKO CLI was already initialized, but completely forgotten and unused.

I will add some basic commands to generate EKOs, useful to test further developments.
- [x] conversion subcommand

### CODE REVIEW

File: src/eko/output/struct.py [hunk 1]
```diff
@@ -10,7 +11,7 @@
 import tarfile
 import tempfile
 import time
-from dataclasses import dataclass, field, fields
+from dataclasses import dataclass
```
Comment (felixhekhorn) [inline src/eko/output/struct.py:14]: since you went for `dataclasses` globally, let's drop also this one

File: src/ekobox/cli/runcards.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+
+@command.group("runcards")
+def subcommand():
+    """Manage EKO runcards."""
+
+
+@subcommand.command("example")
+@option_dest
+def sub_example(destination: pathlib.Path):
```
Comment (felixhekhorn) [inline src/ekobox/cli/runcards.py:25]: - this looks too complicated - we should just create two files and not an additional folder;
- if you want to support custom location (which is fine, but can even be skipped) take two optional arguments `--theory PATH` / `--operator PATH`
- also you should use the constants defined in the other module and drop the `destination` function (which is even inconsistent by the way)

**Review by felixhekhorn — CHANGES_REQUESTED** [review]
I added two v0.8.5 example ekos we for sure want to support given that most of our "real-life" ekos are of this version - both fail:
- the yaml due to the problem given below
- the other with
```
$ poetry run eko convert obf24af_t8e1305.tar
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/felix/.cache/pypoetry/virtualenvs/eko-KkPVjVhh-py3.10/lib/python3.10/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "/home/felix/.cache/pypoetry/virtualenvs/eko-KkPVjVhh-py3.10/lib/python3.10/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "/home/felix/.cache/pypoetry/virtualenvs/eko-KkPVjVhh-py3.10/lib/python3.10/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/felix/.cache/pypoetry/virtualenvs/eko-KkPVjVhh-py3.10/lib/python3.10/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/felix/.cache/pypoetry/virtualenvs/eko-KkPVjVhh-py3.10/lib/python3.10/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/home/felix/Physik/N3PDF/EKO/eko/src/ekobox/cli/convert.py", line 27, in subcommand
    operator = legacy.load_tar(old)
  File "/home/felix/Physik/N3PDF/EKO/eko/src/eko/output/legacy.py", line 278, in load_tar
    metadata = tocard(yaml.safe_load(fd))
  File "/home/felix/Physik/N3PDF/EKO/eko/src/eko/output/legacy.py", line 84, in tocard
    del card["pids"]
KeyError: 'pids'
```

**Comment by felixhekhorn** [comment]
Also benchmarks are broken:
```
$ poe lha
Poe => python benchmarks/lha_paper_bench.py
 ──────────────────────────────────────── 
  Theories: 1 OCards: 1 PDFs: 1 ext: LHA  
 ──────────────────────────────────────── 
Computing for theory=fb3f558, ocard=7f27498 and pdf=ToyLH ...
Traceback (most recent call last):
  File "/home/felix/Physik/N3PDF/EKO/eko/benchmarks/lha_paper_bench.py", line 226, in <module>
    obj.benchmark_plain(0)
  File "/home/felix/Physik/N3PDF/EKO/eko/benchmarks/lha_paper_bench.py", line 123, in benchmark_plain
    self.run_lha(self.plain_theory(pto))
  File "/home/felix/Physik/N3PDF/EKO/eko/benchmarks/lha_paper_bench.py", line 109, in run_lha
    self.run(
  File "/home/felix/.cache/pypoetry/virtualenvs/eko-KkPVjVhh-py3.10/lib/python3.10/site-packages/banana/benchmark/runner.py", line 416, in run
    self.run_config(session, t, o, pdf_name, use_replicas)
  File "/home/felix/.cache/pypoetry/virtualenvs/eko-KkPVjVhh-py3.10/lib/python3.10/site-packages/banana/benchmark/runner.py", line 304, in run_config
    me = self.run_me(t, o, pdf)
  File "/home/felix/Physik/N3PDF/EKO/eko/src/ekomark/benchmark/runner.py", line 119, in run_me
    out = eko.run_dglap(theory, ocard)
  File "/home/felix/Physik/N3PDF/EKO/eko/src/eko/__init__.py", line 29, in run_dglap
    r = runner.Runner(theory_card, operators_card)
  File "/home/felix/Physik/N3PDF/EKO/eko/src/eko/runner.py", line 50, in __init__
    new_theory, new_operators = compatibility.update(theory_card, operators_card)
  File "/home/felix/Physik/N3PDF/EKO/eko/src/eko/compatibility.py", line 57, in update
    new_operators["rotations"]["pids"] = operators["pids"]
KeyError: 'pids'
```

File: src/eko/output/struct.py (hunk 1 above)
Comment (AleCandido) [reply src/eko/output/struct.py:14]: On this I'm plainly inconsistent :D

File: src/ekobox/cli/runcards.py (hunk 2 above)
Comment (AleCandido) [reply src/ekobox/cli/runcards.py:25]: > * if you want to support custom location (which is fine, but can even be skipped) take two optional arguments `--theory PATH` / `--operator PATH`

This is definitely redundant: it makes it more complicate, while you can instead rename them with `mv` after generation.

> * also you should use the constants defined in the other module and drop the `destination` function (which is even inconsistent by the way)

Fine for the constants, I just added them later on. The `destination()` function I implemented too many times, simply here we do not have (yet) many commands that produce outputs, but just two. One is this, and the other one already has a complex enough mixture of inputs and outputs.

In any case, the basic idea is that we support smoothly a basic workflow, providing sensible defaults.
E.g.:
```sh
eko runcards example
mv runcards myoperator
cd myoperator
eko run .
```
it is simple enough and works out of the box.

File: src/ekobox/cli/convert.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+def subcommand(old: pathlib.Path, new: Optional[pathlib.Path]):
+    """Convert old EKO files to new format.
+
+    The OLD file path is used also for the new one, appending "-new" to its
+    stem, unless it is explicitly specified with the dedicated option.
+    """
+    if new is None:
+        new = old.parent / old.with_stem(old.stem + "-new")
+
+    operator = legacy.load_tar(old)
```
[the remark this answers is not in the document]
Comment (AleCandido) [reply src/ekobox/cli/convert.py:27]: The tar one I want to support, and for that I will do all the required validation and branching. YAML is already raising, without me doing anything, that is fine.

**Comment by AleCandido** [comment]
> Also benchmarks are broken:
> 
> ```
> $ poe lha
> Poe => python benchmarks/lha_paper_bench.py
>  ──────────────────────────────────────── 
>   Theories: 1 OCards: 1 PDFs: 1 ext: LHA  
>  ──────────────────────────────────────── 
> Computing for theory=fb3f558, ocard=7f27498 and pdf=ToyLH ...
> Traceback (most recent call last):
>   File "/home/felix/Physik/N3PDF/EKO/eko/benchmarks/lha_paper_bench.py", line 226, in <module>
>     obj.benchmark_plain(0)
>   File "/home/felix/Physik/N3PDF/EKO/eko/benchmarks/lha_paper_bench.py", line 123, in benchmark_plain
>     self.run_lha(self.plain_theory(pto))
>   File "/home/felix/Physik/N3PDF/EKO/eko/benchmarks/lha_paper_bench.py", line 109, in run_lha
>     self.run(
>   File "/home/felix/.cache/pypoetry/virtualenvs/eko-KkPVjVhh-py3.10/lib/python3.10/site-packages/banana/benchmark/runner.py", line 416, in run
>     self.run_config(session, t, o, pdf_name, use_replicas)
>   File "/home/felix/.cache/pypoetry/virtualenvs/eko-KkPVjVhh-py3.10/lib/python3.10/site-packages/banana/benchmark/runner.py", line 304, in run_config
>     me = self.run_me(t, o, pdf)
>   File "/home/felix/Physik/N3PDF/EKO/eko/src/ekomark/benchmark/runner.py", line 119, in run_me
>     out = eko.run_dglap(theory, ocard)
>   File "/home/felix/Physik/N3PDF/EKO/eko/src/eko/__init__.py", line 29, in run_dglap
>     r = runner.Runner(theory_card, operators_card)
>   File "/home/felix/Physik/N3PDF/EKO/eko/src/eko/runner.py", line 50, in __init__
>     new_theory, new_operators = compatibility.update(theory_card, operators_card)
>   File "/home/felix/Physik/N3PDF/EKO/eko/src/eko/compatibility.py", line 57, in update
>     new_operators["rotations"]["pids"] = operators["pids"]
> KeyError: 'pids'
> ```
> 
> EDIT: sorry is already a problem in the base branch ... I'll raise there

**Pull request closed without merging**